File: backend/adapters/api/appointments.py
"""Flask Blueprint for appointments routes."""
from flask import Blueprint, request, jsonify
from logger import get_logger
from adapters.api.helpers import get_current_user_id, require_auth, paginate_list
from services.appointment_service import AppointmentService

appointments_bp = Blueprint('appointments', __name__)

appointment_service = AppointmentService()
logger = get_logger("routes.appointments")

@appointments_bp.route('/appointments', methods=['GET', 'POST'])
def appointments():
    user_id = get_current_user_id()
    if not user_id:
        return jsonify({'success': False, 'errors': ['Não autorizado']}), 401

    if request.method == 'POST':
        data = request.get_json(silent=True) or {}
        data['userId'] = user_id
        result = appointment_service.create_appointment(data)
        if result['success']:
            return jsonify(result), 201
        return jsonify(result), 400

    date_param = request.args.get('date')
    if date_param:
        result = appointment_service.get_appointments_by_date(user_id, date_param)
    else:
        result = appointment_service.get_appointments_by_user_id(user_id)
    return jsonify({'success': True, 'appointments': result}), 200

@appointments_bp.route('/appointments/today', methods=['GET'])
def appointments_today():
    user_id = get_current_user_id()
    if not user_id:
        return jsonify({'success': False, 'errors': ['Não autorizado']}), 401

    result = appointment_service.get_today_appointments(user_id)
    return jsonify({'success': True, 'appointments': result}), 200

@appointments_bp.route('/appointments/upcoming', methods=['GET'])
def appointments_upcoming():
    user_id = get_current_user_id()
    if not user_id:
        return jsonify({'success': False, 'errors': ['Não autorizado']}), 401

    result = appointment_service.get_upcoming_appointments(user_id)
    return jsonify({'success': True, 'appointments': result}), 200

@appointments_bp.route('/appointments/<int:appointment_id>/confirm', methods=['POST'])
def confirm_appointment(appointment_id):
    user_id = get_current_user_id()
    if not user_id:
        return jsonify({'success': False, 'errors': ['Não autorizado']}), 401

    result = appointment_service.confirm_appointment(appointment_id, user_id)
    if result['success']:
        return jsonify(result), 200
    return jsonify(result), 400

@appointments_bp.route('/appointments/<int:appointment_id>/complete', methods=['POST'])
def complete_appointment(appointment_id):
    user_id = get_current_user_id()
    if not user_id:
        return jsonify({'success': False, 'errors': ['Não autorizado']}), 401

    result = appointment_service.complete_appointment(appointment_id, user_id)
    if result['success']:
        return jsonify(result), 200
    return jsonify(result), 400

@appointments_bp.route('/appointments/<int:appointment_id>/cancel', methods=['POST'])
def cancel_appointment(appointment_id):
    user_id = get_current_user_id()
    if not user_id:
        return jsonify({'success': False, 'errors': ['Não autorizado']}), 401

    result = appointment_service.cancel_appointment(appointment_id, user_id)
    if result['success']:
        return jsonify(result), 200
    return jsonify(result), 400

File: frontend-next/src/services/api/appointments.ts
import type { ApiResponse, Appointment } from '@/types';
import { api } from './client';

export const appointmentApi = {
  async create(data: Partial<Appointment>) { return api.post<ApiResponse>('/api/appointments', data); },
  async getByDate(date: string) { return api.get<ApiResponse & { appointments: Appointment[] }>(`/api/appointments?date=${encodeURIComponent(date)}`); },
  async getToday() { return api.get<ApiResponse & { appointments: Appointment[] }>('/api/appointments/today'); },
  async getUpcoming() { return api.get<ApiResponse & { appointments: Appointment[] }>('/api/appointments/upcoming'); },
  async getById(id: number) { return api.get<ApiResponse & { appointment: Appointment }>(`/api/appointments/${id}`); },
  async update(id: number, data: Partial<Appointment>) { return api.put<ApiResponse>(`/api/appointments/${id}`, data); },
  async delete(id: number) { return api.delete<ApiResponse>(`/api/appointments/${id}`); },
  async confirm(id: number) { return api.post<ApiResponse>(`/api/appointments/${id}/confirm`); },
  async complete(id: number) { return api.post<ApiResponse>(`/api/appointments/${id}/complete`); },
  async cancel(id: number) { return api.post<ApiResponse>(`/api/appointments/${id}/cancel`); },
};
